pkg/sql/opt_exec_factory.go

```diff
@@ -1944,21 +1944,22 @@ func (ef *execFactory) ConstructAlterTableSplit(
 		return nil, err
 	}
 
-	knobs := ef.planner.ExecCfg().TenantTestingKnobs
-	if !(knobs != nil && knobs.AllowSplitAndScatter) && !ef.planner.ExecCfg().Codec.ForSystemTenant() {
-		return nil, errorutil.UnsupportedWithMultiTenancy(54254)
-	}
-
 	expirationTime, err := parseExpirationTime(ef.ctx, ef.planner.EvalContext(), expiration)
 	if err != nil {
 		return nil, err
 	}
 
+	reason := roachpb.AdminSplitRequest_Manual
+	knobs := ef.planner.ExecCfg().TenantTestingKnobs
+	if knobs != nil && knobs.AllowSplitAndScatter {
+		reason = roachpb.AdminSplitRequest_Ingestion
```

Comment:
This seems kind of weird to convert `AdminSplitRequest_Manual` to `AdminSplitRequest_Ingestion` as a way to bypass the check in KV, but I guess it could eventually be replaced by giving the test tenant the correct permissions?

Also, by implementing this as a Reason enum instead of a bool, the KV layer is now coupled to the layers above since you need to add new values as new call sites are added.